microphone/drv_audio_pdm.c

```diff
@@ -101,8 +104,18 @@ static void drv_audio_pdm_event_handler(nrfx_pdm_evt_t const * const p_evt)
 			{
 				pdm_buf[l].released = true;
 				data_source_info.audio_source_info.audio_buffer = &pdm_buf[l].mic_buf;
+				if (l == 0 && timestamp_buffer[l] == 0) {
+					// First buffer, set timestamp to recording start timestamp
+					timestamp_buffer[l] = recording_start_timestamp;
+				} else {
+					// Subsequent buffers
+					// t_rec_i = t_release_i - (PDM_BUF_SIZE / effective_sampling_rate) * 1000
+					uint32_t base_sample_rate = 20000; // 20kHz
+					uint32_t effective_sampling_rate = drv_audio_get_mode() == 0 ? base_sample_rate/2 : base_sample_rate;
```

Comment:
Sorry, another question now that I pay more attention to this line. Should it be like this instead? I thought in stereo we have twice the data.
```suggestion
					uint32_t effective_sampling_rate = drv_audio_get_mode() == 0 ? base_sample_rate : base_sample_rate * 2;
```

Reply:
Turns out, it should not be divided or multiplied by 2 at all. The stereo mode records 32 bits per sample (2x16) L+R, and the mono mode records 16 bits per sample, but store 2*L. So, the sample rate should he the same for both mono and stereo.

Reply:
That's not correct. Both are 16 bit samples, Jose will soon merge this PR that fixes that https://github.com/TUDelft-SPC-Lab/midge-code/pull/21/files#diff-62709ef798591e6082a7647c82f7cdea03ade0d8b65b39859fee77d43784178eR30-L23